src/pressure/VCycleMG.cpp

```diff
@@ -22,105 +21,96 @@
-    b_res1->copy_data(b);
+    m_residuum0 = new Field*[m_levels];
+    m_residuum1 = new Field*[m_levels + 1];
+    m_error0 = new Field*[m_levels];
+    m_error1 = new Field*[m_levels + 1];
+    m_mg_temporal_solution = new Field*[m_levels + 1];
 
+    // copies of out and b to prevent aliasing
     // residuum
+    auto *b_res1 = new Field(b);
     b_res1->copyin();
-    residuum1.push_back(b_res1);
+    *(m_residuum1) = b_res1;
 
     // error
+    auto *out_err1 = new Field(out);
     out_err1->copyin();
-    error1.push_back(out_err1);
+    *(m_error1) = out_err1;
 
     // temporal solution
+    auto *out_tmp = new Field(out);
     out_tmp->copyin();
-    mg_temporal_solution.push_back(out_tmp);
-
-    // building Fields for level + sending to GPU
-    // levels going up
-    for (size_t i = 0; i < m_levels; ++i) {
+    *(m_mg_temporal_solution) = out_tmp;
 
-        // build residuum0
-        Field *r0 = new Field(FieldType::P, 0.0, i, domain->get_size(i));
+    // building fields for level + sending to GPU
+    // level going up
+    for (size_t level = 0; level < m_levels; ++level) {
+        // build m_residuum0
+        auto *r0 = new Field(FieldType::P, 0.0, level);
         r0->copyin();
-        residuum0.push_back(r0);
+        *(m_residuum0 + level) = r0;
 
-        // build residuum1
-        Field *r1 = new Field(FieldType::P, 0.0, i + 1, domain->get_size(i + 1));
+        // build m_residuum1
+        auto *r1 = new Field(FieldType::P, 0.0, level + 1);
         r1->copyin();
-        residuum1.push_back(r1);
+        *(m_residuum1 + level + 1) = r1;
 
-        //  build error1
-        Field *e1 = new Field(FieldType::P, 0.0, i + 1, domain->get_size(i + 1));
+        //  build m_error0
+        auto *e0 = new Field(FieldType::P, 0.0, level);
+        e0->copyin();
+        *(m_error0 + level) = e0;
+
+        //  build m_error1
+        auto *e1 = new Field(FieldType::P, 0.0, level + 1);
         e1->copyin();
-        error1.push_back(e1);
+        *(m_error1 + level + 1) = e1;
 
-        // build mg_temporal_solution
-        Field *mg = new Field(FieldType::P, 0.0, i + 1, domain->get_size(i + 1));  // new field to prevent aliasing
+        // build m_mg_temporal_solution
+        auto *mg = new Field(FieldType::P, 0.0, level + 1);
         mg->copyin();
-        mg_temporal_solution.push_back(mg);
-    }
-
-    //  build err0
-    err0.resize(m_levels + 1);
-
-    auto level = error1[0]->get_level();
-    Field *e00 = new Field(FieldType::P, 0.0, level, domain->get_size(level));
-    e00->copyin();
-
-    err0[0] = e00;
-
-    // building Fields for level
-    // levels going down
-    for (int i = m_levels; i > 0; --i) {
-        // build err0
-        level = error1[i - 1]->get_level();
-        Field *e0 = new Field(FieldType::P, 0.0, level, domain->get_size(level));
-        e0->copyin();
-
-        err0[i] = e0;
-    }
+        *(m_mg_temporal_solution + level + 1) = mg;
+    } // end level loop
```

Comment:
two spaces between code and comment are needed.

Reply:
b167247